From 34e22ce597fdd9f1aefb13d93678223623086767 Mon Sep 17 00:00:00 2001
From: Yann Gautier <yann.gautier@foss.st.com>
Date: Wed, 3 Jul 2024 17:19:00 +0200
Subject: [PATCH 648/841] feat(stm32mp2): display package type for STM32MP21

The package type is read from OTP fuse. Display it with other SoC
information at boot.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git 3dc57a773f822e365edcb02a0e9a56d7737e73bd feat(stm32mp2): display package type for STM32MP21]
Change-Id: Ib75461f4eebd56be5e29c93d488c0a8f84d5575a
Signed-off-by: Yann Gautier <yann.gautier@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/tf-a/+/391246
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Maxime MERE <maxime.mere@foss.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
---
 plat/st/stm32mp2/stm32mp2_def.h     |  7 +++++++
 plat/st/stm32mp2/stm32mp2_private.c | 17 +++++++++++++++++
 2 files changed, 24 insertions(+)

diff --git a/plat/st/stm32mp2/stm32mp2_def.h b/plat/st/stm32mp2/stm32mp2_def.h
index 1af082ccb..7dfc9e083 100644
--- a/plat/st/stm32mp2/stm32mp2_def.h
+++ b/plat/st/stm32mp2/stm32mp2_def.h
@@ -117,6 +117,13 @@
 /*******************************************************************************
  * PACKAGE ID
  ******************************************************************************/
+#if STM32MP21
+#define STM32MP21_PKG_CUSTOM		U(0)
+#define STM32MP21_PKG_AL_VFBGA361	U(1)
+#define STM32MP21_PKG_AN_VFBGA273	U(3)
+#define STM32MP21_PKG_AO_VFBGA225	U(4)
+#define STM32MP21_PKG_AM_TFBGA289	U(5)
+#endif /* STM32MP21 */
 #if STM32MP25
 #define STM32MP25_PKG_CUSTOM		U(0)
 #define STM32MP25_PKG_AL_VFBGA361	U(1)
diff --git a/plat/st/stm32mp2/stm32mp2_private.c b/plat/st/stm32mp2/stm32mp2_private.c
index b555398b9..2f8fe2c4f 100644
--- a/plat/st/stm32mp2/stm32mp2_private.c
+++ b/plat/st/stm32mp2/stm32mp2_private.c
@@ -382,6 +382,23 @@ void stm32mp_get_soc_name(char name[STM32_SOC_NAME_SIZE])
 
 	/* Package */
 	switch (get_cpu_package()) {
+#if STM32MP21
+	case STM32MP21_PKG_CUSTOM:
+		pkg = "XX";
+		break;
+	case STM32MP21_PKG_AL_VFBGA361:
+		pkg = "AL";
+		break;
+	case STM32MP21_PKG_AN_VFBGA273:
+		pkg = "AN";
+		break;
+	case STM32MP21_PKG_AO_VFBGA225:
+		pkg = "AO";
+		break;
+	case STM32MP21_PKG_AM_TFBGA289:
+		pkg = "AM";
+		break;
+#endif /* STM32MP21 */
 #if STM32MP25
 	case STM32MP25_PKG_CUSTOM:
 		pkg = "XX";
-- 
2.47.2

